File: ExercismFSharp/ExercismFSharp/BeerSong.fs
module BeerSong

let verse num =
    match num with
    | 0 -> "No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n"
    | 1 -> "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n"
    | 2 -> "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n"
    | _ -> sprintf "%d bottles of beer on the wall, %d bottles of beer.\nTake one down and pass it around, %d bottles of beer on the wall.\n" num num (num - 1)

let verses start stop =
    [ start .. -1 .. stop ]
    |> List.map verse
    |> List.fold (sprintf "%s%s\n") ""

let sing =
    verses 99 0

File: Exercism/python/beer-song/beer_song.py
def recite(start, take=1):
    result = []

    while take > 0:
        result += verse(start)

        start -= 1
        if start < 0:
            start = 99
        take -= 1

        if take > 0:
            result.append("")

    return result


def verse(number):
    if number == 0:
        return [
            "No more bottles of beer on the wall, no more bottles of beer.",
            "Go to the store and buy some more, 99 bottles of beer on the wall."
        ]
    elif number == 1:
        return [
            "1 bottle of beer on the wall, 1 bottle of beer.",
            "Take it down and pass it around, no more bottles of beer on the wall."
        ]
    elif number == 2:
        return [
            "2 bottles of beer on the wall, 2 bottles of beer.",
            "Take one down and pass it around, 1 bottle of beer on the wall."
        ]
    else:
        return [
            "{} bottles of beer on the wall, {} bottles of beer.".format(number, number),
            "Take one down and pass it around, {} bottles of beer on the wall.".format(number - 1)
        ]